packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWDrawer/CWDrawer.tsx

```diff
@@ -0,0 +1,61 @@
+import useBrowserWindow from 'client/scripts/hooks/useBrowserWindow';
+import React, { ComponentProps, useState } from 'react';
+import Drawer from 'react-modern-drawer';
+import 'react-modern-drawer/dist/index.css';
+import { CWIcon } from '../../cw_icons/cw_icon';
+import { CWText } from '../../cw_text';
+import { ComponentType } from '../../types';
+import './CWDrawer.scss';
+
+type CWDrawerProps = Omit<ComponentProps<typeof Drawer>, 'direction'> & {
+  direction?: 'left' | 'right' | 'top' | 'bottom';
+  header: string;
+};
+
+export const CWDrawer = ({
+  direction,
+  duration,
+  open,
+  onClose,
+  children,
+  header,
+}: CWDrawerProps) => {
+  const { isWindowExtraSmall } = useBrowserWindow({});
+  const [fullscreen, setFullscreen] = useState(false);
+
+  const toggleDrawerFullscreen = () => {
+    setFullscreen((prevState) => !prevState);
+  };
+
+  return (
+    <Drawer
+      duration={duration}
+      size={fullscreen ? '100vw' : isWindowExtraSmall ? '90vw' : '50vw'}
+      open={open}
+      onClose={onClose}
+      direction={direction || 'right'}
+      enableOverlay={false}
```

Comment:
should we be able to close the drawer when clicking outside of it, similarly to modals? I would say yes

Reply:
Per conversation with @zakhap and @jessmart1213, drawer should close when clicking outside of it.
Done!